src/laser/Laser.H

```diff
@@ -0,0 +1,63 @@
+#ifndef LASER_H_
+#define LASER_H_
+
+#include <AMReX_MultiFab.H>
+#include <AMReX_Vector.H>
+#include <AMReX_AmrCore.H>
+
+/** \brief describes which slice with respect to the currently calculated is used */
+struct WhichLaserSlice {
+    enum slice { This=0, PrevZeta1, PrevZeta2, PrevTime1, PrevTime2, AbsSq, AbsSqDx, AbsSqDy, N };
+};
+
+class Laser
+{
+private:
+        static constexpr int m_nslices = 8;
+public:
+    /** Constructor */
+    explicit Laser ()
+    {
+        ReadParameters();
+    };
+
+    void ReadParameters ();
+
+    /** get function for the 2D slices
+     * \param[in] islice slice index
+     */
+    amrex::MultiFab& getSlices (int islice) {return m_slices[islice]; }
+
+    /** \brief Allocate beam particle data and initialize particles with requested beam profile
+     * \return physical time at which the simulation will start
+     */
+    void InitData (const amrex::BoxArray& slice_ba,
+                   const amrex::DistributionMapping& slice_dm);
+
+    void PrepareLaserSlice (const amrex::Geometry& geom, const int islice);
+
+    bool m_use_laser {false}; /**< whether a laser is used or not */
+
+    // Same function whether the full fields are specified or just the envelope.
+    // Just somewhere: if (m_EB) { F *= cos (k0 * z) }
+
+    /** Number of guard cells for slices MultiFab */
+    static amrex::IntVect m_slices_nguards;
```

Comment:
I think this member variable does not need to be `static` and could also go in the `private:` section below?
```suggestion
    amrex::IntVect m_slices_nguards = {-1, -1, -1};
```